### ISSUE TITLE
Wrong error message while using check NewlineAtEndOfFile

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#18619]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/newlineatendoffile.html#NewlineAtEndOfFile
I have downloaded the latest checkstyle from: https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-13.0.0
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat -A a.txt
This file has line ending CR^MThis file as new line at end of file but with CR^M
```

File as line ending CR.

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="NewlineAtEndOfFile">
    <property name="lineSeparator" value="lf"/>
  </module>
</module>

```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.0.0-all.jar -c config.xml a.txt
Starting audit...
[ERROR] /home/atharv/test/a.txt:1: File does not end with a newline. [NewlineAtEndOfFile]
Audit done.
Checkstyle ends with 1 errors.
```

Error expected was:
`Expected line ending for file is LF(\\n), but CR(\\r) is detected.`

Similar kind of example is used https://checkstyle.sourceforge.io/checks/misc/newlineatendoffile.html#Example3-code

CR is considered valid line separator: https://checkstyle.sourceforge.io/property_types.html#LineSeparatorOption

**Comment by Saadmadni84** [comment]
Hi maintainers,
I would like to work on this issue and submit a fix for the incorrect error message when a wrong line separator is detected. Please let me know if I can proceed.

**Comment by romani** [comment]
we can improve message to be `File does not end with a newline 'lf'.`

This check should not care what symbols are in this files, and what symbol is at of file already. Main point that it is not 'lf'.

**Comment by romani** [comment]
@Saadmadni84 ,  issue is approve now.

but if you are new to our project please [read](https://github.com/checkstyle/checkstyle/blob/master/.github/CONTRIBUTING.md#getting-started) and watch videos at [Starting_Development](https://checkstyle.org/beginning_development.html#Starting_Development).

**Comment by Avanish-Gupta-CSE** [comment]
I am on it.

**Comment by Anushreebasics** [comment]
@Atharv3221 can I work on this issue?

**Comment by Atharv3221** [comment]
Ping @VidytBhudolia are you still working on this ?

**Comment by mdhamed238** [comment]
Hi @romani,

I reproduced the issue locally using the released jar:

```bash
$ java -jar checkstyle-13.2.0-all.jar -c a_config.xml a.txt
Starting audit...
[ERROR] .../a.txt:1: File does not end with a newline. [NewlineAtEndOfFile]
Audit done.
```

Where `a.txt` ends with a bare CR (`\r`) and `config.xml` sets `lineSeparator = lf`.

---

Based on your feedback — *"the check should not care what symbol is already at end of file, main point is that it is not 'lf'"* — I'm proposing the following approach:

**Parameterize the existing `noNewlineAtEOF` message** to include the expected separator, rather than detecting what the file actually ends with:

```properties
# before
noNewlineAtEOF=File does not end with a newline.

# after
noNewlineAtEOF=File does not end with a newline ''{0}''.
```

The `readAndCheckFile` logic stays simple — no additional detection branches:

```java
else if (!endsWithNewline(randomAccessFile, lineSeparator)) {
    log(1, MSG_KEY_NO_NEWLINE_EOF,
            lineSeparator.toString().toLowerCase(Locale.ENGLISH));
}
```

This produces the output:

```bash
$ java -jar checkstyle-13.2.0-SNAPSHOT-all.jar -c a_config.xml a.txt
Starting audit...
[ERROR] .../a.txt:1: File does not end with a newline 'lf'. [NewlineAtEndOfFile]
Audit done.
```

The change also covers other `lineSeparator` values consistently (e.g. `'cr'`, `'crlf'`).

Could you confirm this is the direction you'd like, so I can finalize the PR?

**Comment by mdhamed238** [comment]
I'm on it.

**Comment by mahfouz72** [comment]
fix is merged at https://github.com/checkstyle/checkstyle/pull/18972
